model/DataObjects.inc.php

```diff
@@ -628,6 +628,13 @@ public static function delete($libraryID, $key) {
 			Zotero_FullText::deleteItemContent($obj);
 		}
 		
+		$storageFileID = null;
+		if ($type == 'item' && $obj->isImportedAttachment()) {
+			// Get storageFileID while it still exists in storageFileItems table
+			$info = Zotero_Storage::getLocalFileItemInfo($obj);
+			$storageFileID = $info['storageFileID'];
+		}
+		
```

Comment:
Combine with previous section (with just an `isImportedAttachment` test, since we're already testing for `$type` and `isAttachment`)